### ISSUE TITLE
Colors missing when opening Octo buffers directly

### ISSUE THREAD

**Issue by williambdean** [pwntester/octo.nvim#771]
[The harpoon plugin](https://github.com/ThePrimeagen/harpoon/tree/harpoon2) allows you to pin files in a picker, which persists across Neovim restarts.

I've noticed that after a restart, opening octo buffers through harpoon - or even just with `:edit octo://...` some syntax highlighting is missing, e.g., the diff bar and the assignees aren't coloured.

If I open the Octo picker with the partially coloured PR open and close it again, everything is coloured correctly.
So, I assume opening Octo - not just loading it via Lazy - triggers some additional functionality that I'm not calling when opening an Octo PR without going through Octo itself.

Does anyone have experience using octo with harpoon or have some pointers how I could address this?

_Originally posted by @paulhdk in https://github.com/pwntester/octo.nvim/discussions/649_

**Comment by wd60622** [comment]
Don't think this is limited to harpoon. 

using `:e <octo-buffer-uri>` from fresh start has the same issue

**Comment by paulhdk** [comment]
@wd60622 in #649:
> I experience this as well. I am thinking the highlight groups arent initialized or something.
Have you played around with it more and discovered anything?

I haven't. But I want to give it another go when I have more time. This could make a nice first contribution from my side, I hope!

**Comment by wd60622** [comment]
It'd be a great contribution, indeed! 
Well toy around with it a bit. If you do have any information, comment it here in this thread. And it will get solved eventually 😄

**Comment by GuillaumeLagrange** [comment]
Are you guys lazy loading Octo in any way ?

**Comment by paulhdk** [comment]
> Are you guys lazy loading Octo in any way ?

Yes. I have `lazy = true` and load Octo through the mappings I've set.

**Comment by wd60622** [comment]
I have lazy loading as well. But did try without which still had the same effect

**Comment by wd60622** [comment]
It seems like the colors just aren't setup before the buffer is opened / on the first open with the buffer autocmds.


```lua
-- Call this first
require "octo.ui.colors".setup()
```

Would you be able to give this a try and see if it fixes the issues? @paulhdk

**Comment by williambdean** [comment]
Do you manually do that edit or do you provide a hook with harpoon? @vimscientist69

**Comment by vimscientist69** [comment]
> Do you manually do that edit or do you provide a hook with harpoon? [@vimscientist69](https://github.com/vimscientist69)

Just created a pull request. 

https://github.com/ThePrimeagen/harpoon/pull/665

**Comment by vimscientist69** [comment]
@williambdean , not sure if the error problem you encounter is the exact as mine (fixed in the pull request). Just thought it would be relevant to share. 

If you want to use it without waiting for the PR to get merged, use my fork: https://github.com/vimscientist69/harpoon

**Comment by paulhdk** [comment]
> It seems like the colors just aren't setup before the buffer is opened / on the first open with the buffer autocmds.
> 
> -- Call this first
> require "octo.ui.colors".setup()
> --- Then open the buffer with the uri
> 
> Would have to have this in one of the autocmds as well
> 
> [octo.nvim/lua/octo/commands.lua](https://github.com/pwntester/octo.nvim/blob/8217c8d95faec659c4f32a85cb5a6a3f1d584ca5/lua/octo/commands.lua?plain=1#L867-L870)
> 
> Lines 867 to 870 in [8217c8d](/pwntester/octo.nvim/commit/8217c8d95faec659c4f32a85cb5a6a3f1d584ca5)
> 	 if not _G.octo_colors_loaded then 
> 	   colors.setup() 
> 	   _G.octo_colors_loaded = true 
> 	 end 
> 
> Would you be able to give this a try and see if it fixes the issues? [@paulhdk](https://github.com/paulhdk)

Hm. I'm not able to reproduce the problem right now. 

When I try to access a PR after a nvim reboot, it doesn't load anything (probably because the Octo plugin isn't loaded yet?), and when I load Octo, e.g., by opening the picker, and then try to access a PR again, it works fine, colours and all.

**Comment by vimscientist69** [comment]
> > It seems like the colors just aren't setup before the buffer is opened / on the first open with the buffer autocmds.
> > -- Call this first
> > require "octo.ui.colors".setup()
> > --- Then open the buffer with the uri
> > Would have to have this in one of the autocmds as well
> > [octo.nvim/lua/octo/commands.lua](https://github.com/pwntester/octo.nvim/blob/8217c8d95faec659c4f32a85cb5a6a3f1d584ca5/lua/octo/commands.lua?plain=1#L867-L870)
> > Lines 867 to 870 in [8217c8d](/pwntester/octo.nvim/commit/8217c8d95faec659c4f32a85cb5a6a3f1d584ca5)
> > if not _G.octo_colors_loaded then
> > colors.setup()
> > _G.octo_colors_loaded = true
> > end
> > Would you be able to give this a try and see if it fixes the issues? [@paulhdk](https://github.com/paulhdk)
> 
> Hm. I'm not able to reproduce the problem right now.
> 
> When I try to access a PR after a nvim reboot, it doesn't load anything (probably because the Octo plugin isn't loaded yet?), and when I load Octo, e.g., by opening the picker, and then try to access a PR again, it works fine, colours and all.

That happens to me as well. To solve this, execute `Octo pr` if you are checked out on that PR branch. If not, open the PR with `Octo pr list`. 

I have a mapping `<leader>opo` that executes `Octo pr` to open the current pr when I close and re-open neovim. I interpret it as "octo pr open".

**Comment by williambdean** [comment]
Can you both give this a try

https://github.com/pwntester/octo.nvim/pull/985

**Comment by vimscientist69** [comment]
> edit octo://pwntester/octo.nvim/issue/771

It works. Is there a Octo command to open a pull request or issue from any repository (an alternative to using edit octo://username/reponame/pull or issue/number?

**Comment by vimscientist69** [comment]
Is there a command or a way to list all notifications, including ones that I've opened in Github inbox? When I run `Octo notification list`, I get the following. Also, it seems like Octo only shows notifications for repositories that I am an owner of, or repositories in organizations that I am a member of.

<img width="1352" alt="Image" src="https://github.com/user-attachments/assets/1b48f3c9-53fd-46fc-bef6-8cbd815e527c" />

In Github, I have some unresolved notifications.

<img width="1352" alt="Image" src="https://github.com/user-attachments/assets/696d3010-d837-486e-b27c-e4144d01f97a" />

**Comment by williambdean** [comment]
@vimscientist69 you have to use a custom command to get all notifications. 
I have it here: 
https://github.com/williambdean/dotfiles/blob/81e9b1338abc4fd4438baff0f2046f092216ffb8/nvim/lua/plugins/version-control.lua?plain=1#L359-L369

**Comment by williambdean** [comment]
As for the edit, I think it has to be done with `Octo <url>` or `Octo octo://<buffer-name>`. There might be a short hand but I'm not sure.

**Comment by vimscientist69** [comment]
The notifications command works as expected. Thanks :)

**Comment by vimscientist69** [comment]
URL works. Thanks, now I can use octo to manage issues an PRs on repositories without cloning and navigating to them first. Truly thanks.
